src/physics/negative_spalart_allmaras_rans_model.cpp

```diff
@@ -0,0 +1,718 @@
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+template <typename real2>
+std::array<real2,nstate-(dim+2)> ReynoldsAveragedNavierStokes_SAneg<dim,nstate,real>
+::compute_effective_viscosity_turbulence_model_templated (
+    const std::array<real2,dim+2> &primitive_soln_rans,
+    const std::array<real2,nstate-(dim+2)> &primitive_soln_turbulence_model) const
+{   
+    const real2 laminar_dynamic_viscosity = this->navier_stokes_physics->compute_viscosity_coefficient(primitive_soln_rans);
+    const real2 laminar_kinematic_viscosity = laminar_dynamic_viscosity/primitive_soln_rans[0];
+
+    const real2 coefficient_f_n = this->compute_coefficient_f_n(primitive_soln_turbulence_model[0],laminar_kinematic_viscosity);
+
+    std::array<real2,nstate-(dim+2)> effective_viscosity_turbulence_model;
+
+    for(int i=0; i<nstate-(dim+2); ++i){
+        if constexpr(std::is_same<real2,real>::value){
+            effective_viscosity_turbulence_model[i] = (laminar_dynamic_viscosity+coefficient_f_n*primitive_soln_rans[0]*primitive_soln_turbulence_model[0])/sigma;
+        }
+        else if constexpr(std::is_same<real2,FadType>::value){
+            effective_viscosity_turbulence_model[i] = (laminar_dynamic_viscosity+coefficient_f_n*primitive_soln_rans[0]*primitive_soln_turbulence_model[0])/sigma_fad;
+        }
+        else{
+            std::cout << "ERROR in physics/negative_spalart_allmaras_rans_model.cpp --> compute_effective_viscosity_turbulence_model_templated(): real2 != real or FadType" << std::endl;
+            std::abort();
+        }
+        effective_viscosity_turbulence_model[i] = scale_coefficient(effective_viscosity_turbulence_model[i]);
+    }
+    
+    return effective_viscosity_turbulence_model;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+template<typename real2>
+real2 ReynoldsAveragedNavierStokes_SAneg<dim,nstate,real>
+::compute_coefficient_Chi (
+    const real2 nu_tilde,
+    const real2 laminar_kinematic_viscosity) const
+{
+    // Compute coefficient Chi
+    const real2 Chi = nu_tilde/laminar_kinematic_viscosity;
+
+    return Chi;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+template<typename real2>
+real2 ReynoldsAveragedNavierStokes_SAneg<dim,nstate,real>
+::compute_coefficient_f_v1 (
+    const real2 coefficient_Chi) const
+{
+    // Compute coefficient f_v1
+    real2 coefficient_f_v1;
+
+    if constexpr(std::is_same<real2,real>::value){ 
+        coefficient_f_v1 = coefficient_Chi*coefficient_Chi*coefficient_Chi/(coefficient_Chi*coefficient_Chi*coefficient_Chi+c_v1*c_v1*c_v1);
+    }
+    else if constexpr(std::is_same<real2,FadType>::value){ 
+        coefficient_f_v1 = coefficient_Chi*coefficient_Chi*coefficient_Chi/(coefficient_Chi*coefficient_Chi*coefficient_Chi+c_v1_fad*c_v1_fad*c_v1_fad);
```

Comment:
`coefficient_Chi*coefficient_Chi*coefficient_Chi` pre-compute for efficiency

Reply:
fixed.
sorry I should check carefully